activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -63,8 +63,8 @@ def locking_enabled? #:nodoc:
 
         def increment_lock
           lock_col = self.class.locking_column
-          previous_lock_value = send(lock_col).to_i
-          send(lock_col + "=", previous_lock_value + 1)
+          previous_lock_value = _read_attribute(lock_col)
```

Comment:
Yes. I agree with removing the `to_i` but I don't think using `read_atttribute` make it more consistent. It actually may break application and the fact that we are using `__send__` instead of `write_attribute` make it more inconsistent.